FirebaseAuth/Tests/SampleSwift/AuthenticationExample/CustomViews/LoginView.swift

```diff
@@ -12,162 +12,145 @@
+//      } catch let error as AuthErrorCode
+//        where error.code == .secondFactorRequired {
+//        // error as? AuthErrorCode == nil because AuthErrorUtils returns generic
+//        /Errors
+//        // https://firebase.google.com/docs/auth/ios/totp-mfa#sign_in_users_with_a_second_factor
+      } catch let error as NSError
+        where error.code == AuthErrorCode.secondFactorRequired.rawValue {
+        let mfaKey = AuthErrorUserInfoMultiFactorResolverKey
+        guard let resolver = error.userInfo[mfaKey] as? MultiFactorResolver else { throw error }
+        await MainActor.run {
+          dismiss()
+          delegate?.loginDidOccur(resolver: resolver)
+        }
+      } catch {
+        print(error.localizedDescription)
+      }
+    }
   }
 
-  private func setupEmailTextfield() {
-    emailTextField = textField(placeholder: "Email", symbolName: "person.crop.circle")
-    emailTextField.translatesAutoresizingMaskIntoConstraints = false
-    addSubview(emailTextField)
-    NSLayoutConstraint.activate([
-      emailTextField.leadingAnchor.constraint(
-        equalTo: safeAreaLayoutGuide.leadingAnchor,
-        constant: 15
-      ),
-      emailTextField.trailingAnchor.constraint(
-        equalTo: safeAreaLayoutGuide.trailingAnchor,
-        constant: -15
-      ),
-      emailTextField.heightAnchor.constraint(equalToConstant: 45),
-    ])
-
-    let constant: CGFloat = UIDevice.current.orientation.isLandscape ? 15 : 50
-    emailTopConstraint = emailTextField.topAnchor.constraint(
-      equalTo: safeAreaLayoutGuide.topAnchor,
-      constant: constant
-    )
-    emailTopConstraint.isActive = true
+  private func createUser() {
+    Task {
+      do {
+        _ = try await AppManager.shared.auth().createUser(
+          withEmail: email,
+          password: password
+        )
+        // Sign-in was successful.
+        await MainActor.run {
+          dismiss()
+          delegate?.loginDidOccur(resolver: nil)
+        }
+      } catch {
+        // TODO(ncooke3): Implement error display.
+        print(error.localizedDescription)
+      }
+    }
   }
+}
 
-  private func setupPasswordTextField() {
-    passwordTextField = textField(placeholder: "Password", symbolName: "lock.fill")
-    passwordTextField.translatesAutoresizingMaskIntoConstraints = false
-    addSubview(passwordTextField)
-    NSLayoutConstraint.activate([
-      passwordTextField.leadingAnchor.constraint(
-        equalTo: safeAreaLayoutGuide.leadingAnchor,
-        constant: 15
-      ),
-      passwordTextField.trailingAnchor.constraint(
-        equalTo: safeAreaLayoutGuide.trailingAnchor,
-        constant: -15
-      ),
-      passwordTextField.heightAnchor.constraint(equalToConstant: 45),
-    ])
-
-    let constant: CGFloat = UIDevice.current.orientation.isLandscape ? 5 : 20
-    passwordTopConstraint =
-      passwordTextField.topAnchor.constraint(
-        equalTo: emailTextField.bottomAnchor,
-        constant: constant
+extension LoginView {
```

Comment:
<img width="405" alt="Screenshot 2024-12-12 at 11 28 29 AM" src="https://github.com/user-attachments/assets/63a32a67-c28c-45c9-a7be-5100fc44651a" />

Reply:
Putting the UI code in an extension to keep all the UI code in one place.

Reply:
Resolving so I can merge.